content/docs/user-guide/scripting/script-canvas/programmer-guide/behavior-context.md

```diff
@@ -4,91 +4,241 @@ description: Learn about the important relationship between Script Canvas and th
+These events are part of the `LightComponentRequestBus`. Their behavior is implemented by `LightComponent`.
+
+```cpp
+bool LightComponent::TurnOnLight()
+{
+    bool success = m_light.TurnOn();
+    if (success)
+    {
+        LightComponentNotificationBus::Event(GetEntityId(), &LightComponentNotifications::LightTurnedOn);
+    }
+    return success;
+}
+
+bool LightComponent::TurnOffLight()
+{
+    bool success = m_light.TurnOff();
+    if (success)
+    {
+        LightComponentNotificationBus::Event(GetEntityId(), &LightComponentNotifications::LightTurnedOff);
+    }
+    return success;
+}
+
+void LightComponent::ToggleLight()
+{
+    if (m_light.IsOn())
+    {
+        TurnOffLight();
+    }
+    else
+    {
+        TurnOnLight();
+    }
+}
+```
 
-The following code shows the binding for the `BehaviorLightComponentNotificationBusHandler` and defines two events: `LightTurnedOn` and `LightTurnedOff`.
+To make these events accessible for scripting, their methods must be reflected to the behavior context. This is done in `LightComponent::Reflect`.
+
+```cpp
+if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+{
+    behaviorContext->EBus<LightComponentRequestBus>("Light", "LightComponentRequestBus")
+        ->Attribute(AZ::Script::Attributes::Category, "Rendering")
+        ->Event("TurnOn", &LightComponentRequestBus::Events::TurnOnLight)
+        ->Event("TurnOff", &LightComponentRequestBus::Events::TurnOffLight)
+        ->Event("Toggle", &LightComponentRequestBus::Events::ToggleLight);
+}
+```
+
+When Script Canvas examines the behavior context, it finds these events and automatically generates the corresponding nodes.
+
+![Light component nodes in Script Canvas](/images/user-guide/scripting/script-canvas/behavior-context-ebus-request-light-nodes.png)
+
+#### EBus events and EntityID
+
+EBuses communicate with an entity's components. To do this, it needs an address. All component EBuses derive from `AZ::ComponentBus`, which is addressable by an ID of the type `AZ::EntityId`. For this reason, all nodes from a component EBus have a data pin for an `EntityID`. The presence of `Self` in the `EntityID` field refers to the `EntityID` of the entity that owns the Script Canvas graph. However, this ID can be assigned to another entity, or even changed to an invalid entity ID.
+
+![Self EntityID](/images/user-guide/scripting/script-canvas/behavior-context-ebus-request-entityID.png)
+
+#### Tooltips
+
+Script Canvas nodes should include a helpful tooltip for every parameter. For example, the Light component might have a `SetLightState` event with a `state` parameter:
+
+```cpp
+// Set the light state to on or off.
+void SetLightState(State state);
+```
+
+You should add a tooltip in the behavior context reflection to describe the parameter. In this example, the tooltip will show "1=On, 0=Off" when a user hovers over the `State` parameter on the `SetState` node.
```

Comment:
```suggestion
You should add a tooltip in the behavior context reflection to describe the parameter. In this example, the tooltip will show `1=On, 0=Off` when a user hovers over the `State` parameter on the `SetState` node.
```

Reply:
Updated the comment to use bold for nodes and node pins and parameters. There is precedence for this in other parts of the Script Canvas section.